src/main/java/io/mathdojo/Question.java

```diff
@@ -63,29 +101,92 @@ public void setHints(String[] hints) {
 		this.hints = hints;
 	}
 
-	public int getSuccessRate() {
-		return successRate;
+	public String getParentTopicTitle() {
+		return parentTopicTitle;
 	}
 
-	public void setSuccessRate(int successRate) {
-		this.successRate = successRate;
+	public void setParentTopicTitle(String parentTopicTitle) {
+		this.parentTopicTitle = parentTopicTitle;
 	}
 
-	public String getDifficulty() {
-		return difficulty;
+	public String[] getQuestionAnswerOptions() {
+		return questionAnswerOptions;
 	}
 
-	public void setDifficulty(String difficulty) {
-		this.difficulty = difficulty;
+	public void setQuestionAnswerOptions(String[] questionAnswerOptions) {
+		this.questionAnswerOptions = questionAnswerOptions;
+	}
+
+	public String getAnswer() {
+		return answer;
 	}
 
-	public boolean isSolved() {
-		return solved;
+	public void setAnswer(String answer) {
+		this.answer = answer;
 	}
 
-	public void setSolved(boolean solved) {
-		this.solved = solved;
+	@Override
+	public int hashCode() {
+		final int prime = 31;
+		int result = 1;
+		result = prime * result + ((id == null) ? 0 : id.hashCode());	
+		return result;
 	}
 
+	@Override
+	public boolean equals(Object obj) {
+		if (this == obj)
+			return true;
+		if (obj == null)
+			return false;
+		if (getClass() != obj.getClass())
+			return false;
+		Question other = (Question) obj;
+		if (answer == null) {
+			if (other.answer != null)
+				return false;
+		} else if (!answer.equals(other.answer))
+			return false;
+		if (difficulty == null) {
+			if (other.difficulty != null)
+				return false;
+		} else if (!difficulty.equals(other.difficulty))
+			return false;
+		if (!Arrays.equals(hints, other.hints))
+			return false;
+		if (id == null) {

```

Comment:
Minor detail, need not be corrected now be this could be:
```suggestion
		if (id == null && other.id != null) {
                        return false
```

Reply:
that's an eclipse generated equals method. I can come up with a better format at some point with less if and return statements, but I'll leave that for now